library/src/main/java/nl/joery/animatedbottombar/AnimatedBottomBar.kt

```diff
@@ -472,7 +474,19 @@ class AnimatedBottomBar @JvmOverloads constructor(
         if (viewPager != null) {
             selectTabAt(viewPager.currentItem, false)
             viewPager.addOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+                private var previousState:Int = ViewPager.SCROLL_STATE_IDLE
+                private var userScrollChange = false
                 override fun onPageScrollStateChanged(state: Int) {
+                    // Use Scroll state to detect whether the user is sliding
+                    if (previousState == ViewPager.SCROLL_STATE_DRAGGING
```

Comment:
Minor issue, but to match the rest of the code, I would add curly braces around all if statements.